pruner/light/pruner.go

```diff
@@ -3,15 +3,30 @@ package light
 import (
 	"context"
 
+	"github.com/ipfs/boxo/blockservice"
+
 	"github.com/celestiaorg/celestia-node/header"
+	"github.com/celestiaorg/celestia-node/share/ipld"
 )
 
-type Pruner struct{}
+type Pruner struct {
+	bserv blockservice.BlockService
+}
 
-func NewPruner() *Pruner {
-	return &Pruner{}
+func NewPruner(bserv blockservice.BlockService) *Pruner {
+	return &Pruner{bserv: bserv}
 }
 
-func (p *Pruner) Prune(context.Context, *header.ExtendedHeader) error {
+func (p *Pruner) Prune(ctx context.Context, h *header.ExtendedHeader) error {
+	var roots [][]byte
+	roots = append(roots, h.DAH.RowRoots...)
+	roots = append(roots, h.DAH.ColumnRoots...)
+	for _, root := range roots {
+		cid := ipld.MustCidFromNamespacedSha256(root)
+		if err := ipld.DeleteNode(ctx, p.bserv, cid); err != nil {
+			return err
+		}
+	}
```

Comment:
Architectural note:
I don't think Pruners should be accessing low-level storage primitives and instead we should have some form of `share.Deleter` in between.

Reply:
Yeah, thats why i said `share.Deleter`